src/app/core/auth/services/auth.service.ts

```diff
@@ -35,6 +36,17 @@ export class AuthService {
   }
 
   public getUserProfile(): Observable<User> {
-    return this._http.get<User>('/api/me')
+    return this._http.get<User>('/api/me').pipe(
+      map(data => {
+        return {
+          ...data,
+          avatar: environment.apiUrl + data.avatar
+        }
+      })
+    )
+  }
+
+  public updateUserInfo(user: FormData): Observable<User> {
+    return this._http.patch<User>('/api/settings', user)
   }
```

Comment:
I have a strong feeling that these two methods should be extracted into a separate `ProfileService`. Auth and Profile are two different concerns, so should not be included in the same class. That would be violation of Single Responsibility Principle